### PR TITLE
Default Generator Status

### PR DESCRIPTION
**Pull request by tiffanyperumpail**
Change to DefaultGenerator:
When the generate() method throw an exception, the exception is caught and a stack trace is printed. Due to the way this exception is caught, it is difficult to detect if code generation has failed when we aren't generating code through the command line. 

For this reason, I added enum CodeGenStatus, so a status can be set for each instance of the DefaultGenerator.

This change is inspired by our needs at Wikia, where we are using Swagger to generate code progammatically and would like to easily detect when generation fails.

Please let us know if we've missed anything or didn't take any factors into account with this change.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -456,7 +456,7 @@
         </repository>
     </repositories>
     <properties>
-        <swagger-parser-version>1.0.9-SNAPSHOT</swagger-parser-version>
+        <swagger-parser-version>1.0.9</swagger-parser-version>
```
Comment (wing328) [inline pom.xml]: Thanks for the PR. May I know why you need to use swagger parser 1.0.9 instead of 1.0.9-SNAPSHOT ?

Comment (tiffanyperumpail) [inline pom.xml]: My changes don't depend on using swagger parser 1.0.9 instead of 1.0.9-SNAPSHOT; I made that change because the Travis CI failed to build until I bumped the version of swagger-parser to 1.0.9. I could change it back if necessary.

screen shot 2015-08-06 at 10 48 44 am

Comment (who) [inline pom.xml]: @wing328 There appears to be a problem with one of the dependencies, even with a fresh, unmodified clone. I filed a bug here: https://github.com/swagger-api/swagger-codegen/issues/1055

Comment (fehguy) [inline pom.xml]: I have no idea why that snapshot was deleted--I've redeployed it so 1.0.9 should now be available.

**Comment by wing328** [comment]
@hgwood would this PR meet your requirement to better catch errors for DefaultGenerator? (ref: #993)

**Comment by hgwood** [comment]
Yes it would. FYI, [here's](https://github.com/restlet/swagger-codegen/commit/253e95c911e8032cbfe4a896572e6bbaf84b0f46) what we currently have at Restlet.

However, there are [many other places](https://github.com/swagger-api/swagger-codegen/search?utf8=%E2%9C%93&q=printStackTrace) in swagger-codegen where exceptions are swallowed. So I don't trust it to indicate failure even after this PR is merged.

To really fix #993 without too much work, *all* `e.printStackTrace` could be replaced by `throw RuntimeException(e)`, and try-catch blocks could be added in main classes (`io.swagger.codegen.SwaggerCodegen` and `io.swagger.codegen.Codegen`), where `e.printStackTrace` is expected.

Alternatively, another system of error handling could be put in place, maybe with checked exceptions, enum return values (like in this PR) or monads.

In conclusion, although this PR is interesting and probably useful in the short term, it does not fix the whole problem. I ultimately think the dev team needs to decide how to handle errors project-wide. Once that's done, I'd be willing to help with the implementation, and I'm sure @tiffanyperumpail would be too.
